components/PoeModal.tsx

```diff
@@ -0,0 +1,659 @@
+                        );
+                      }
+
+                      const methodSchema = new Schema(
+                        contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      );
+
+                      const value = contract.methodsObject[
+                        detail.parameters.entrypoint
+                      ](
+                        methodSchema.Execute(detail.parameters.value)
+                      ).toTransferParams().parameter?.value;
+
+                      if (!value) return undefined;
+
+                      const param = emitMicheline(value as Expr);
+
+                      const parser = new Parser();
+
+                      const type = emitMicheline(
+                        parser.parseJSON(
+                          contract.entrypoints.entrypoints[
+                            detail.parameters.entrypoint
+                          ]
+                        ),
+                        {
+                          indent: "",
+                          newline: "",
+                        }
+                      );
+
+                      return {
+                        type: "contract",
+                        values: {
+                          lambda: generateExecuteContractMichelson(version, {
+                            address: detail.destination,
+                            amount: Number(detail.amount),
+                            entrypoint: detail.parameters.entrypoint,
+                            type,
+                            param,
+                          }),
+                          metadata: null,
+                        },
+                      };
+                    } catch (e) {
+                      const errorMessage = `Failed to create a TzSafe proposal: ${
+                        (e as Error).message
+                      }`;
+                      setTransactionError(errorMessage);
+                      console.log("Contract conversion error:", e);
+
+                      state.p2pClient?.sendError(
+                        message.id,
+                        errorMessage,
+                        BeaconErrorType.TRANSACTION_INVALID_ERROR
+                      );
+                      return undefined;
+                    }
+                  } else
+                    return {
+                      type: "transfer",
+                      values: {
+                        to: detail.destination,
+                        amount: detail.amount,
+                        parameters: {},
+                      },
+                    };
+
+                case TezosOperationType.DELEGATION:
+                  return !!detail.delegate
+                    ? {
+                        type: "lambda",
+                        values: {
+                          lambda: generateDelegateMichelson(version, {
+                            bakerAddress: detail.delegate,
```

Comment:
check address is valid address at least.
(nice to check if it's backer.)

Reply:
![image](https://github.com/marigold-dev/tzsafe-ui/assets/115862/cc7251a8-f94a-409f-880c-be97acea858e)